include/deal.II/grid/grid_tools.h

```diff
@@ -1145,6 +1145,77 @@ namespace GridTools
    const FullMatrix<double>                                          &matrix = FullMatrix<double>(),
    const std::vector<unsigned int>                                   &first_vector_components = std::vector<unsigned int>());
 
+  /*@}*/
+  /**
+   * @name Dealing with boundary and manifold ids
+   */
+  /*@{*/
+
+  /**
+   * Copy boundary ids to manifold ids. The default manifold_id for
+   * new Triangulation objects is numbers::invalid_manifold_id. When
+   * refinements occurs, the Triangulation asks where to locate new
+   * points to the underlying manifold, and if the manifold_id of a
+   * boundary face is set to numbers::invalid_manifold_id, then
+   * Triangulation reverts back boundary_id (this was the behavior of
+   * deal.II < 8.2). This function copies the boundary_ids of the
```

Comment:
I would remove this discussion of behavior. It is documented elsewhere and if we ever change it this comment will become stale. I would restrict the documentation to what the function does and, if appropriate to a rationale.